pgmpy/models/DoubleMLRegressor.py

```diff
@@ -0,0 +1,188 @@
+        treatment_col: Optional[str] = None,
+        n_folds: int = 5,
+        random_state: Optional[int] = None,
+    ):
+        self.dag = dag
+        self.estimator_g = estimator_g
+        self.estimator_m = estimator_m
+        self.adjustment_set = adjustment_set
+        self.treatment_col = treatment_col
+        self.n_folds = n_folds
+        self.random_state = random_state
+
+    def _more_tags(self):
+        return {"requires_y": True, "no_sparse_input": True}
+
+    # ---------- small helpers ----------
+    def _as_dataframe(self, X) -> pd.DataFrame:
+        if isinstance(X, pd.DataFrame):
+            return X.copy()
+        arr = np.asarray(X)
+        if arr.ndim == 1:
+            arr = arr.reshape(-1, 1)
+        cols = [f"x{i}" for i in range(arr.shape[1])]
+        return pd.DataFrame(arr, columns=cols)
+
+    def _resolve_treatment_and_adjustments(
+        self, df: pd.DataFrame
+    ) -> Tuple[str, List[str]]:
+        # treatment priority: explicit -> dag.exposure -> first column
+        if self.treatment_col is not None and self.treatment_col in df.columns:
+            treat = self.treatment_col
+        else:
+            exp = getattr(self.dag, "exposure", None)
+            if exp is None:
+                treat = df.columns[0]
+            else:
+                exp_list = list(exp) if isinstance(exp, (list, tuple, set)) else [exp]
+                if len(exp_list) != 1:
+                    raise NotImplementedError(
+                        "Only single exposure supported in " "skeleton."
+                    )
+                treat = exp_list[0] if exp_list[0] in df.columns else df.columns[0]
+
+        # adjustments priority: explicit -> dag.get_adjustment_set() -> []
+        if self.adjustment_set is not None:
+            adj = [c for c in self.adjustment_set if c in df.columns and c != treat]
+        else:
+            adj = []
+            if hasattr(self.dag, "get_adjustment_set"):
+                try:
+                    adj = [
+                        c
+                        for c in self.dag.get_adjustment_set()
+                        if c in df.columns and c != treat
+                    ]
+                except Exception:
+                    adj = []
+        return treat, adj
+
+    def fit(self, X, y, sample_weight: Optional[np.ndarray] = None):
+
+        X_arr, y_arr = check_X_y(
+            X, y, accept_sparse=False, ensure_2d=True, force_all_finite=True
+        )
+
+        X = X if isinstance(X, pd.DataFrame) else self._as_dataframe(X_arr)
+        feature_names = list(X.columns)
+
+        self.n_features_in_ = X_arr.shape[1]
+        self.feature_names_in_ = np.array(feature_names, dtype=object)
+
+        df = X.copy()
+        df["outcome"] = np.asarray(y_arr).ravel()
+
+        treat_col, adj_cols = self._resolve_treatment_and_adjustments(df)
+        design_cols = [treat_col] + [c for c in adj_cols if c != treat_col]
```

Comment:
There should never be any common variables between `treatment` and `adjustment`.